src/main/java/com/team766/controllers/CanSparkMaxSmartMotionPIDController.java

```diff
@@ -163,49 +159,61 @@ public void setCLEPosition(double position){
 			position = minpos1;
 		}
 
-		pid1.setReference(position, ControlType.kSmartMotion);
+		pid.setReference(position, ControlType.kSmartMotion);
 	}
 	//1st step to go to a position using the normal PID, setting what you want the position to be
-	public void setPosition(double position){
+	public void setSetpoint(double position){
 		if(position > maxpos1){
 			position = maxpos1;
 		} else if(position < minpos1){
 			position = minpos1;
 		}
 
-		currentPos = position;
+		setPointPosition = position;
+
+		theState = PIDSTATE.PID;
 	}
 
 	public void stop(){
 		//Failsafe
-		currentPos = mc1.getSensorPosition();
+		setPointPosition = mc.getSensorPosition();
 		theState = PIDSTATE.OFF;
 	}
 
 	//run loop that actually runs the PID using normal dz
 	//You need to call this function repedatly in OI
 	public void run(boolean enabled){
 		if(enabled){
+			//Checking if Abs encoder is enabled, and if so we wouldn't want positions above 1 and below 0
+			if(isAbsoluteEncoderEnabled){
+				MathUtil.clamp(minPos, 0.0, 1.0);
+				MathUtil.clamp(maxPos, 0.0, 1.0);
+			}
+			
 			switch(theState){
 				case OFF:
 					break;
 				case ANTIGRAV:
-					antigrav();
+					if (mc.getSensorPosition() <= (dz1 + mc.getSensorPosition()) && mc.getSensorPosition() >= (mc.getSensorPosition() - dz1)){
```

Comment:
these conditions are always true if `dz1 >= 0`. I think some of the `mc.getSensorPosition()` should be `setPointPosition`